code/src/components/UserNameInput.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import labyrinth, { getLabyrinth } from 'reducers/labyrinth';
+import { SubmitBtn } from 'library/buttons';
+
+const UserNameInput = ({ onStartGame }) => {
+  const [userNameInputValue, setUserNameInputValue] = useState('');
+  const dispatch = useDispatch();
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    // save the username to global state
+    dispatch(labyrinth.actions.setUserName(userNameInputValue));
+
```

Comment:
Global state was a wise choice for the user name, and excellent comments that made us understand the thinking and coding swiftly.